app/src/controllers/reservation/repositories/ReservationRepository.ts

```diff
@@ -117,6 +117,37 @@ const ReservationRepository: ReservationServiceSocket = {
     })
     return Boolean(reservation)
   },
+  async fetchReservationsByDSS(shopId, startDate, endDate, stylistId) {
+    const reservation = await prisma.reservation.findMany({
+      where: {
+        reservationDate: {
+          gte: startDate,
+          lt: endDate,
+        },
+        AND: { stylistId, shopId },
+      },
+      include: {
+        menu: true,
+        shop: true,
+      },
+    })
+    const reservationData = reservation.map(r => {
+      const datesAndDuration = {
+        reservationDate: r.reservationDate,
+        duration: r.menu.duration,
+      }
+      return datesAndDuration
+    })
+    return reservationData
+  },
+
+  async fetchMenuDuration(menuId) {
+    const menu = await prisma.menu.findUnique({
+      where: { id: menuId },
+    })
+    const duration = menu?.duration
+    return duration || 0
+  },
```

Comment:
0 duration will not exist so you should just return null instead. then throw a not found error in the service layer.